SPIE2016-9913-16/9913-16.tex

```diff
@@ -0,0 +1,303 @@
+
+\subsection{LSST Measurement Algorithms}
+
+There are a number of measurement algorithms in LSST that could be made available as Astropy affiliate packages\cite{bosch_2016_48435}.
+The initial idea is to release the LSST equivalents of a subset of the Astromatic suite\cite{2012ASSP...29...71B}.
+This would have the advantage of providing a simple way to test the process for migrating native LSST measurement algorithm code to the Astropy ecosystem in a form that is easily validatable by the community.
+Once complete it should then be possible to consider migrating the more advanced LSST algorithms.
+Making them available in this way should aid adoption by the wider community and provide early feedback as to the quality of the algorithms and detect any bugs or anomalous behavior well before commissioning of LSST begins.
+
+\subsection{NDData}
+
+The Astropy \texttt{NDData} class is an implementation of the common astronomy data structure\cite{2015A&C....12..146J} of a data array, mask and uncertainty with attached metadata.
+LSST has an \texttt{Exposure} class (one for each underlying data type) represented by a \texttt{MaskedImage} and structured metadata.
+The class layout of an \texttt{Exposure} is shown in Fig.~\ref{fig:exposure}.
+It would be beneficial if an LSST \texttt{Exposure} could implement the \texttt{NDData} interface.
+The mask in an \texttt{Exposure} is a bit mask and not a Boolean mask and the uncertainties are always represented as variances (although covariance has been discussed).
+These are supportable as sub-classes and variants of them are already available in the Astropy ecosystem.
+One item that is currently missing from the Astropy arena is the notion of a pixel origin.
+In LSST the pixel origin can be used to specify the coordinates of the data array relative to a parent data array, such as allowing the coordinates of image cutouts to reference where they came from in the parent.
+LSST nomenclature is to use \emph{PARENT} to indicate the abstract pixel coordinate and \emph{LOCAL} to indicate that the coordinates are relative to the 0-based origin of the data array itself\footnote{The Starlink nomenclature\cite{2015A&C....12..146J} is \emph{PIXEL} and \emph{GRID} where \emph{GRID} follows the FITS convention of starting at 1.}.
+This attachment of a data array to a parent pixel coordinate system is a critical item and we plan to work with Astropy on making this functionality available to \texttt{NDData} users.
+
+\begin{figure} [t]
+\begin{center}
+\includegraphics[width=\textwidth]{exposure-dependencies}
+\end{center}
+\caption[layers]
+%>>>> use \label inside caption to get Fig. number with \ref{}
+{\label{fig:exposure}
+The LSST classes that are used to define a \texttt{lsst.afw.image.Exposure}.}
+\end{figure}
```

Comment:
On the figure:
 - Template parameters here add more confusion than clarity - just use Mask instead of MaskU, etc.
 - ImageOrigin is just an enum, and doesn't need to be here.
 - Mask doesn't inherit from or have an Image; probably best to just drop that relationship entirely rather than include ImageBase.
 - Quadrupole.Matrix should just be Eigen::Matrix2d.
 - Psf.Image should just be Image
 - Use of both periods and :: is confusing
 - AffineXYTransform in cameraGeom.Detector should just be XYTransform or AffineTransform; AffineXYTransform is just an adaptor class.
 - Point2DVector in cameraGeom.Detector is just a typedef to std::vector<Point2D>, right?  Probably don't need that here.
 - Probably just use Function2D in Kernel instead of Function2DList (it's another typedef to STL container).